go.mod

```diff
@@ -28,6 +28,7 @@ require (
 	github.com/huin/goupnp v1.0.3
 	github.com/jackpal/gateway v1.0.6
 	github.com/jackpal/go-nat-pmp v1.0.2
+	github.com/leanovate/gopter v0.2.9
```

Comment:
The property testing library I picked. There are other using generics, but I understand gopter is currently the most stable one